src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -73,6 +73,36 @@ public static void ResetColor()
             }
         }
 
+        public static int WindowWidth
+        {
+            get
+            {
+                if (!ConsoleOutIsTerminal) return 0; // Should we rather throw exception here? On windows we get IOException from the native code, do we need to match for code portability?
+                return TerminalBasicInfo.Instance.ColumnFormat;
```

Comment:
> By native shim do you mean the System.Native?

Yes

> I have not yet tested it by using Environment.GetEnvironmentVariable though

I don't think that'll work:
https://github.com/dotnet/coreclr/issues/635

> Also, I was thinking we could try to read the environment variable COLUMNS, which does get updated on resizing the terminal

That could be an ok fallback to try in the case of ioctl/TIOCGWINSZ not being available or not working.